alchemist/alchemist-implementationbase/src/main/kotlin/it/unibo/alchemist/model/implementations/geometry/DoubleExtension.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) 2010-2020, Danilo Pianini and contributors
+ * listed in the main project's alchemist/build.gradle.kts file.
+ *
+ * This file is part of Alchemist, and is distributed under the terms of the
+ * GNU General Public License, with a linking exception,
+ * as described in the file LICENSE in the Alchemist distribution's top directory.
+ */
+
+package it.unibo.alchemist.model.implementations.geometry
+
+import org.danilopianini.lang.MathUtils
+import java.lang.IllegalArgumentException
+
+/**
+ * Maps a value in [min, max] to [newMin, newMax].
+ */
+fun Double.map(min: Double, max: Double, newMin: Double, newMax: Double): Double = when {
```

Comment:
What does it mean?

Reply:
Maps a value falling in range [min, max] to range [newMin, newMax], is this clearer?

Reply:
No, I do not understand what it does. Can you provide an example? So maybe I can come up with some name

Reply:
This is similar to [Arduino's map](https://www.arduino.cc/reference/en/language/functions/math/map/), it's a linear interpolation. Suppose you have an analog reading (= a value in [0, 1023] in Arduino boards) and you want to map it in the range [0, 180], representing the movement degree of a servomotor. So for instance 0 is mapped to 0, 1023 is mapped to 180 and 512 is mapped to 90.
An alternative name for the function is `linearlyInterpolate`.

Reply:
Ah! Now I get it!
Please do not use `map` alone, it really makes people think of the `map` operation on traversables or of `let`.
It's not a linear interpolation either.

What you want to do is a *scaling* operation. Proposal: use `rescaleFromTo`. If you want it to be beautiful you may do something like:

```kotlin
internal data class DoubleInRange(val value: Double, val range: ClosedFloatingPointRange<Double>) {
   infix fun rescaledTo(newRange: ClosedFloatingPointRange<Double>) = TODO()
}
infix fun Double.within(range: ClosedFloatingPointRange<Double>) = DoubleInRange(this, range) // Plus check if it's in range and fail otherwise
// You should then be able to write something like
1 within 0..2 rescaledTo 2..4 // should return 3
```

Reply:
Very nice suggestion. However, I noticed that function is unused 😅 I'll remove it.